darts/tests/utils/test_timeseries_generation.py

```diff
@@ -368,69 +370,133 @@ def helper_routine(idx, attr, vals_exp, **kwargs):
+            ("W", "weekofyear", 52),
+        ],
+    )
+    def test_datetime_attribute_timeseries_one_hot(self, config):
+        base_freq, attribute_freq, period = config
+        # first month/year, week/year, day/week, hour/day, second/hour
+        simple_start = pd.Timestamp("2000-01-03")
+        idx = generate_index(start=simple_start, length=period, freq=base_freq)
+        vals = np.eye(period)
+
+        # simple start
+        self.helper_routine(idx, attribute_freq, vals_exp=vals, one_hot=True)
+        # with time-zone
+        if attribute_freq == "hour":
+            # shift to mimic conversion from UTC to CET
+            vals = np.roll(vals, shift=-1, axis=0)
+        self.helper_routine(idx, attribute_freq, vals_exp=vals, tz="CET", one_hot=True)
+
+        # missing values
+        cut_period = period // 3
+        idx = generate_index(
+            start=pd.Timestamp("2000-01-03"), length=cut_period, freq=base_freq
+        )
+        vals = np.eye(period)
+        # removing missing rows
+        vals = vals[:cut_period]
+        # mask missing attribute values
+        vals[:, cut_period:] = 0
+
+        self.helper_routine(idx, attribute_freq, vals_exp=vals, one_hot=True)
+
+        # shifted time index
+        shifted_start = pd.Timestamp("2000-03-02 03:00:03")
+        # 3rd month/year, day/week, hour/day, second/hour
+        shift = 3
+        # 9th week of year
+        if attribute_freq == "weekofyear":
+            shift = 9
+
+        # 1-indexed attributes, the columns are shifted by one
+        if attribute_freq in ONE_INDEXED_FREQS:
+            shift -= 1
+
+        idx = generate_index(start=shifted_start, length=period, freq=base_freq)
+        vals = np.eye(period)
+        # shift values
+        vals = np.roll(vals, shift=-shift, axis=0)
+
+        # shifted start
+        self.helper_routine(idx, attribute_freq, vals_exp=vals, one_hot=True)
+
+    @pytest.mark.parametrize("config", [("h", "hour", 24), ("M", "month", 12)])
+    def test_datetime_attribute_timeseries_cyclic(self, config):
+        base_freq, attribute_freq, period = config
+        idx = generate_index(
+            start=pd.Timestamp("2000-01-01"), length=2 * period, freq=base_freq
+        )
+
+        if attribute_freq in ONE_INDEXED_FREQS:
+            shift = 1
+        else:
+            shift = 0
 
-        # ===> one hot encoded
-        # month
-        vals = [1] + [0] * 11
-        vals = [vals for _ in range(48)]
-        helper_routine(idx, "month", vals_exp=vals, one_hot=True)
-
-        # tz=CET, month
-        vals = [1] + [0] * 11
-        vals = [vals for _ in range(48)]
-        helper_routine(idx, "month", vals_exp=vals, tz="CET", one_hot=True)
-
-        # ===> sine/cosine cyclic encoding
-        # hour (period = 24 hours in one day)
-        period = 24
         freq = 2 * np.pi / period
-        vals_dta = [i for i in range(24)] * 2
+        vals_dta = [i for i in range(shift, period + shift)] * 2
```

Comment:
Actually, thinking about this: For cyclic encodings, shouldn't we make all freqs start at 0. So that also for example the first month (month=1) start at sin(0), cos(0), same as the first hour (hour=0) starts at sin(0), cos(0)

Reply:
I agree with you, would be better if all the attributes are "synchronized"

Reply:
👍 We should mention this is in the changelog, since it'll end up generating different predictions (probably)